docs/db-migration-guide.ipynb

```diff
@@ -0,0 +1,137 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Charmed Kubeflow Databases Migration Guide\n",
+    "\n",
+    "There are two databases in Charmed Kubeflow - `katib-db` and `kfp-db`. In Charmed Kubeflow 1.7 these MariaDB databases deployed by `charmed-osm-mariadb-k8s` charm. Since release 1.7 there is an option to relate charm to new MySQL database that is deployed by `mysql-k8s` charm. In cases where previous deployment has MariaDB a migration of database is required."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
```

Comment:
We may need to specify that the migration instructions could be slightly different if using juju 2.9 and 3.1

Reply:
I suspect we will be going with Juju 2.9, since that's what we used to deploy 1.7. MariaDB only exists in <=1.7.

Reply:
So the idea will be to first migrate then upgrade juju, in the case users require it?

Reply:
This should be mentioned in overall CKF guide (non existent at this point): DB migration shoud be done prior to CKF upgrade.

Reply:
I think it's worth mentioning with a big note saying that this step should be performed before even upgrading.

Reply:
Yes. And also need the same in upgrade guide.